src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -134,13 +177,131 @@ private void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode pare
             }
 
             parent.SubFields = subFields;
+
+            fields.Clear();
+            System.Threading.Interlocked.CompareExchange(ref context.ReusableFields, fields, null);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="FieldType"/> for the specified AST <see cref="Field"/> within a specified parent
+        /// output graph type within a given schema. For meta-fields, returns the proper meta-field field type.
+        /// </summary>
+        protected virtual FieldType GetFieldDefinition(ISchema schema, IObjectGraphType parentType, Field field)
```

Comment:
Moved from `ExecutionHelper`; no changes.  I think it belongs here, unless there is a reason this method would be needed outside of the execution strategy.

Reply:
Why virtual?

Reply:
The point of this PR is to allow modifications to any part of the execution strategy, regardless of the reason.

What if we needed to override the meta field to properly support the `@oneOf` directive (specifically, the new `oneField` field on `__Type`)?  That doesn't meet spec, but may be something someone wants to support.  With this change, you could change what graph type is returned without changing any other part of the code.

I'm not saying it is necessary in that case, but it is a good example of a situation where could be necessary to change this method.